storage/postgres/abstract.go

```diff
@@ -111,12 +111,21 @@ func listWithLabelsByCriteria(ctx context.Context, db pgDB, baseEntity interface
 	if err != nil {
 		return nil, err
 	}
+
+	// Lock the rows if we are in transaction so that update operations on those rows can rely on unchanged data
+	// This allows us to handle concurrent updates on the same rows by executing them sequentially as
+	// before updating we have to anyway select the rows and can therefore lock them
+	if _, ok := db.(*sqlx.Tx); ok {
+		sqlQuery = sqlQuery[:len(sqlQuery)-1]
+		sqlQuery += fmt.Sprintf(" FOR SHARE of %s;", baseTableName)
```

Comment:
I also thought about that - currently it will wait  for db query timeout duration and generally update executions will be sequencial which is convenient but failfast might be better in some cases. So generally im not sure which is better